hs/src/Reach/AST/Base.hs

```diff
@@ -309,27 +312,37 @@ type SLPart = B.ByteString
 render_sp :: SLPart -> Doc
 render_sp = viaShow
 
+type UIntTy = Bool
+uintWord :: UIntTy
+uintWord = False
+uint256 :: UIntTy
+uint256 = True
+
+uint256_Max :: Integer
+uint256_Max = 2 ^ (256 :: Integer) - 1
+
 data PrimOp
-  = ADD
-  | SUB
-  | MUL
-  | DIV
-  | MOD
-  | PLT
-  | PLE
-  | PEQ
-  | PGE
-  | PGT
+  = ADD UIntTy
+  | SUB UIntTy
+  | MUL UIntTy
+  | DIV UIntTy
+  | MOD UIntTy
+  | PLT UIntTy
+  | PLE UIntTy
+  | PEQ UIntTy
+  | PGE UIntTy
+  | PGT UIntTy
+  | UCAST UIntTy UIntTy
   | IF_THEN_ELSE
   | DIGEST_EQ
   | ADDRESS_EQ
   | TOKEN_EQ
   | SELF_ADDRESS SLPart Bool Int
   | LSH
```

Comment:
Does UInt256 not support LSH/RSH because Algo? 

TIL `<<` and `>>` aren't in the op code docs

Reply:
Correct.